docs/admin/etcd_upgrade.md

```diff
@@ -0,0 +1,209 @@
+In either case, if high availability is
+required--as it might be in a production cluster--the data directory ought to be
+[backed up periodically](https://coreos.com/etcd/docs/latest/op-guide/recovery.html)
+to reduce downtime in case of corruption.
+
+
+## Background
+
+As of Kubernetes version 1.5.1, we are still using etcd from the 2.2.1 release with
+the v2 API. Also, we have no pre-existing process for updating etcd, as we have
+never updated etcd by either minor or major version.
+
+Note that we need to migrate both the etcd versions that we are using (from 2.2.1
+to at least 3.0.x) as well as the version of the etcd API that Kubernetes talks to. The etcd 3.0.x
+binaries support both the v2 and v3 API.
+
+This document describes how to do this migration.
+
+### etcd upgrade requirements
+
+There are requirements on how an etcd cluster upgrade can be performed. The primary considerations are:
+- Upgrade between one minor release at a time
+- Rollback supported through additional tooling
+
+#### One minor release at a time
+
+Upgrade only one minor release at a time. For example, we cannot upgrade directly from 2.1.x to 2.3.x.
+Within patch releases it is possible to upgrade and downgrade between arbitrary versions. Starting a cluster for 
+any intermediate minor release, waiting until the cluster is healthy, and then
+shutting down the cluster down will perform the migration. For example, to upgrade from version 2.1.x to 2.3.y, 
+it is enough to start etcd in 2.2.z version, wait until it is healthy, stop it, and then start the
+2.3.y version.
+
+#### Rollback via additional tooling
+
+Versions 3.0+ of etcd do not support general rollback. That is,
+after migrating from M.N to M.N+1, there is no way to go back to M.N.
+The etcd team has provided a [custom rollback tool](https://github.com/kubernetes/kubernetes/tree/master/cluster/images/etcd/rollback)
+but the rollback tool has these limitations:
+
+* This custom rollback tool is not part of the etcd repo and does not receive the same
+  testing as the rest of etcd.  We are testing it in a couple of end-to-end tests.
+  There is only community support here.
+
+* The rollback can be done only from the 3.0.x version (that is using the v3 API) to the
+  2.2.1 version (that is using the v2 API).
+
+* The tool only works if the data is stored in `application/json` format.
+
+* Rollback doesn’t preserve resource versions of objects stored in etcd.
+
+The last bullet means that any component or user that has some logic
+depending on resource versions may require restart after etcd rollback. This
+includes that all clients using the watch API, which depends on
+resource versions. Since both the kubelet and kube-proxy use the watch API, a
+rollback might require restarting all Kubernetes components on all nodes.
+
+_To verify: It seems that both Kubelet and KubeProxy are using “resource
+version” only for watching (i.e. are not using resource versions for anything
```

Comment:
Is this text under "To verify" meant to be in the document? It almost looks like a review comment.